### ISSUE TITLE
Create-only actions in OpenSearch bulk requests

### ISSUE THREAD

**Issue by dlvenable** [opensearch-project/data-prepper#1457]
**Is your feature request related to a problem? Please describe.**

Data Prepper only uses the `index` action of the [Bulk API](https://opensearch.org/docs/latest/opensearch/rest-api/document-apis/bulk/). In some cases, teams want to create only if the document does not already exist. The OpenSearch `_bulk` API does support this via a `create` action.

**Describe the solution you'd like**

Create a new `action` property of the OpenSearch sink. The default value will be `index` (this is the current behavior). Pipeline authors can define this value to be either `create` or `index`. 

Data Prepper will send bulk requests with all events using the configured `action`.

If OpenSearch responds that the document could not be created because it exists, this document will be dropped without any retry or DLQ. This is a reasonable expectation for pipeline authors who uses create since their goals are to avoid updating existing documents.

**Describe alternatives you've considered (Optional)**

Data Prepper could key off of values directly in the document to determine if `create` or `index` should be used. This would be somewhat more complicated since pipeline authors would need to define conditions and fallbacks. This approach could be added later if needed. But, the proposed solution should be simpler for pipeline authors.

**Additional context**

In addition to `index` and `create`, the Bulk API also supports `update` and `delete`. The proposal above would allow us to add these in future iterations.

**Comment by jzonthemtn** [comment]
@dlvenable Happy to take on this one if it's ready to be worked.

**Comment by dlvenable** [comment]
@jzonthemtn , Yes, this one is ready to work when you are available. I'll assign it to you. Thanks!

**Comment by jzonthemtn** [comment]
Hi @dlvenable, I'm working on integration tests for this. It looks like a running OpenSearch is required for the `OpenSearchSinkIT` tests. The `OpenSearchIntegrationHelper` gets the host names via the `tests.opensearch.host` system property. I just wanted to verify my understanding here is correct before getting too deep into it.

**Comment by dlvenable** [comment]
Thanks @jzonthemtn ! Yes, you would need to run OpenSearch. You can do this via Docker.

I normally run them using the GitHub Actions as a template.

Start Docker:

https://github.com/opensearch-project/data-prepper/blob/d737dafb39d8f61b7948f086d3ae1951e52be22f/.github/workflows/opensearch-sink-os-integration-tests.yml#L26-L27

Run the tests:

https://github.com/opensearch-project/data-prepper/blob/d737dafb39d8f61b7948f086d3ae1951e52be22f/.github/workflows/opensearch-sink-os-integration-tests.yml#L31-L32
